frontend/test/resolution/testCatch.cpp

```diff
@@ -319,7 +324,535 @@ static void test9() {
+  auto path = UniqueString::get(ctx, "test16.chpl");
+
+  ErrorGuard guard(ctx);
+  std::string program = R""""(
+                        proc canThrow(i: bool): bool throws {
+                          if i then
+                            throw new owned Error();
+
+                          return i;
+                        }
+                        canThrow(true);
+              )"""";
+  ctx->advanceToNextRevision(false);
+  setFileText(ctx, path, program);
+  const ModuleVec& vec = parseToplevel(ctx, path);
+  auto mod = vec[0]->toModule();
+  assert(mod);
+  auto modRes = resolveModule(ctx, mod->id());
+  auto canThrowFn = mod->stmt(0)->toFunction();
+  assert(canThrowFn);
+  auto call = mod->stmt(1)->toFnCall();
+  assert(call);
+  assert(!guard.realizeErrors());
+}
+
+// check relaxed error handling mode
+static void test17(Parser* parser) {
+
+  auto ctx = parser->context();
+  auto path = UniqueString::get(ctx, "test17.chpl");
+
+  ErrorGuard guard(ctx);
+  std::string program = R""""(
+                        module M {
+                          proc canThrow(i: bool): bool throws {
+                            if i then
+                              throw new owned Error();
+
+                            return i;
+                          }
+                          canThrow(true);
+                        }
+              )"""";
+  ctx->advanceToNextRevision(false);
+  setFileText(ctx, path, program);
+  const ModuleVec& vec = parseToplevel(ctx, path);
+  auto mod = vec[0]->toModule();
+  assert(mod);
+  auto call = mod->stmt(1)->toFnCall();
+  assert(call);
+  auto modRes = resolveModule(ctx, mod->id());
+  auto canThrowFn = mod->stmt(0)->toFunction();
+  assert(canThrowFn);
+  assert(guard.numErrors() == 1);
+  assert(guard.error(0)->message() == "call to throwing function 'canThrow' without throws, try, or try! (relaxed mode)");
+  assert(guard.error(0)->kind() == ErrorBase::Kind::ERROR);
+  assert(guard.realizeErrors() == 1);
+}
+
+// check fatal error handling mode prototype module
+static void test18(Parser* parser) {
+
+  auto ctx = parser->context();
+  auto path = UniqueString::get(ctx, "test18.chpl");
+
+  ErrorGuard guard(ctx);
+  std::string program = R""""(
+                        prototype module M {
+                          proc canThrow(i: bool): bool throws {
+                            if i then
+                              throw new owned Error();
+
+                            return i;
+                          }
+                          canThrow(true);
+                        }
+              )"""";
+  ctx->advanceToNextRevision(false);
+  setFileText(ctx, path, program);
+  const ModuleVec& vec = parseToplevel(ctx, path);
+  auto mod = vec[0]->toModule();
+  assert(mod);
+  auto call = mod->stmt(1)->toFnCall();
+  assert(call);
+  auto modRes = resolveModule(ctx, mod->id());
+  auto canThrowFn = mod->stmt(0)->toFunction();
+  assert(canThrowFn);
+
+  assert(!guard.realizeErrors());
```

Comment:
As David has once informed me, I believe this is handled by the ErrorGuard's destructor already.